controller/lib/hal/psol.cpp

```diff
@@ -89,8 +100,17 @@ void HalApi::PSOL_Value(float val) {
 
   val = std::clamp(val, 0.0f, 1.0f);
 
-  // We get maximum volume at about 50% duty cycle
-  float duty = val * static_cast<float>(tmr->reload);
+  // Scale linearly between our fully-open and fully-closed PWM values, except
+  // to be a little paranoid, we say "0 means 0".
```

Comment:
This is being conservative in the direction of closing the psol, i.e. providing no oxygen. Should we also be conservative in the opposite direction, i.e. giving full oxygen?

Reply:
Hm, that's an interesting idea.

I think we actually should not do that.  psol fully open doesn't mean "give me 100% oxygen", but "give me full pressure of oxygen".  Which might be quite high, e.g. 10psi, which is on the low end for what we're expecting, is 700 cmH2O.

If somehow our experiment was wrong and 0.90 is not "fully open", then I don't think we'd want to have a discontinuity in the pressure we're delivering and potentially deliver 10psi when PID is expecting us to deliver much less.

Reply:
(Whereas the possible fully-closed discontinuity doesn't hurt as much.  The worst thing that happens is we don't deliver enough air, and that's an alarm, but it's not shooting 10psi through the system and blowing it up.)

Reply:
Sounds reasonable.